### ISSUE TITLE
"Graphing Days" doesn't change graphs

### ISSUE THREAD

**Issue by shawnp0wers** [MPOS/php-mpos#2432]
I've read through the graphing refactoring threads, and thought I understood how it works. However, if I change the "Graphing Days" setting in the admin page from 1 to say, 30 -- I don't see any changes.

My concern is that I don't actually have the historical data to pull from, and that it might be purged based on the previous "1 day" setting. Am I out of luck regarding past data, or have I completely misunderstood the graphing changes?

Basically, my boss wants me to show him the hashrates over the past 30 days. If the data isn't in the pool database, I'm out of luck.

**Comment by TheSerapher** [comment]
It's fetching from cache. Clear your mencache and try again. Fetches data from user statistics table so it should be there.

**Comment by shawnp0wers** [comment]
Hmm, no effect (waited to make sure cron ran).

My graphs still show " Average Hashrate past 24h" -- should that change to 30 days, or am I misunderstanding?

NOTE: Running commit from Feb 7, 2015

**Comment by TheSerapher** [comment]
Template has 24h hard coded on it. Should probably be updated to show the current setting. Data should be a month though - at least as far as I can tell from the function. There isn't a lot of magic in it and like said before, it does fetch from the user stats table:

https://github.com/MPOS/php-mpos/blob/master/include/classes/statistics.class.php#L701

**Comment by shawnp0wers** [comment]
It looks like my statistics_users table only has 1550 records in it. I'm guessing it was auto-pruned at 1 day perhaps? (ie, before I changed it to 30 days... it'll probably grow now)

Crap.

But thank you for the help. :)

**Comment by TheSerapher** [comment]
Of course you are right, forgot about the cleanup job. It will grow to your setting now, but all previous data is lost.
